robot-name/Pratster95/robot_name.go

```diff
@@ -0,0 +1,40 @@
+package robotname
+
+import (
+	"fmt"
+	"math/rand"
+)
+
+// Define the Robot type here.
+type Robot struct {
+	name string
+}
+
+var nameUsed = map[string]bool{}
+
+const max = 26 * 26 * 10 * 10 * 10
+
+func (r *Robot) Name() (string, error) {
+	if r.name != "" {
+		return r.name, nil
+	}
+	if len(nameUsed) >= max {
+		return "", fmt.Errorf("No Names left")
+	}
+	for {
+		name := getName()
+		if !nameUsed[name] {
```

Comment:
I had no idea that the default value is returned for those non-existent keys. Nice.

I would still use the usual syntax for checking if the element is present though. It would be more consistent as it wouldn't change the condition depending in the values' type.